armasec/token_security.py

```diff
@@ -153,3 +138,56 @@ async def __call__(self, request: Request) -> TokenPayload:
                 )
 
         return token_payload
+
+    def _load_all_managers(self) -> None:
+        for domain_config in self.domain_configs:
+            try:
+                manager = self._load_manager(domain_config)
+                self.managers.append(manager)
+            except AuthenticationError:
+                self.debug_logger(f"Failed to match JWK against domain {domain_config.domain}")
+            except Exception as err:
+                if self.debug_exceptions:
+                    self.debug_logger(f"Exception caught: {err.__class__.__name__}")
+
+    def _load_manager(self, domain_config: DomainConfig) -> TokenManager:
+        self.debug_logger(f"Lazy loading TokenManager for domain {domain_config.domain}")
+        loader = OpenidConfigLoader(domain_config.domain, debug_logger=self.debug_logger)
+        decoder = TokenDecoder(loader.jwks, domain_config.algorithm, debug_logger=self.debug_logger)
+        return TokenManager(
+            loader.config,
+            decoder,
+            audience=domain_config.audience,
+            debug_logger=self.debug_logger,
+        )
+
+    def _extract_token_payload_from_manager(self, request: Request) -> TokenPayload:
+        token_payload = None
+
+        if len(self.managers) == 0:
+            self._load_all_managers()
+
+        AuthenticationError.require_condition(
+            len(self.managers) > 0,
+            "Not authenticated: coudln't load any TokenManager instance",
+        )
+
+        for manager in self.managers:
+            try:
+                token_payload = manager.extract_token_payload(request.headers)
+                break
+            except Exception as err:
+                if self.debug_exceptions:
```

Comment:
We don't need this check here.

The `debug_exceptions` setting should probably be renamed to something like `raise_original_exceptions` or something like that. The intent here is this:

When we are debugging our App's usage of Armasec, and a token is rejected for whatever reason, we would like Armasec to tell us details about why the token was rejected. It does this by raising specific exception types with details about the issue.

When our App that uses Armasec is in production, we want Armasec to give a simple "good" or "bad" when we ask it to authenticate a request. We never want any of the details about why the token wasn't accepted to be able to get back to the user. So, when `debug_exceptions` is not set, Armasec will capture any raised exceptions at the outermost level and instead return a simple 401 or 403 response (as appropriate).

So, the setting is poorly named and it should indicate that better.

Reply:
should the code just pass so?